app/forms/provider_form.rb

```diff
@@ -0,0 +1,28 @@
+class ProviderForm
+  include ActiveModel::Model
+
+  attr_accessor :code
+
+  validates :code, presence: true
+  validate :provider_already_onboarded?
+
+  def onboard
+    return false unless valid?
+
+    provider.update!(placements: true)
+  end
+
+  def provider
+    @provider ||= Provider.find_by(code:)
+  end
+
+  private
+
+  def provider_already_onboarded?
+    if provider.blank?
+      errors.add(:code, :blank)
+    elsif provider.placements?
+      errors.add(:code, :already_added, provider_name: provider.name)
+    end
+  end
```

Comment:
Ditto

Reply:
resolved in [8a29568](https://github.com/DFE-Digital/itt-mentor-services/pull/105/commits/8a2956808a0e940e50960406e039c4df9a898787)